src/bootstrap/test.rs

```diff
@@ -1072,6 +1072,8 @@ note: if you're sure you want to do this, please open an issue as to why. In the
 
         cmd.arg("--docck-python").arg(builder.python());
 
+        cmd.arg("--jsondocck-path").arg(builder.ensure(tool::JsonDocCk { compiler, target }));
```

Comment:
I found the issue: `compiler` here should always be the bootstrap compiler. You're trying to compile it with stage 1. Try using stage 0 instead.

Reply:
How would I get the stage0 compiler here? I see above it seems to use the same compiler for rustdoc, so I'm not sure how to get the older one.

Reply:
I made it use stage0, I also made jsondocck only build when testing rustdoc-json to match rustdoc's behavior in compiletest